src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,339 @@
+            if (!string.IsNullOrEmpty(legalNoticeCaption) || !string.IsNullOrEmpty(legalNoticeText))
+            {
+                warningReasons.Add(StringUtil.Loc("UITestingWarning_LegalNotice"));
+            }
+
+            //auto-logon
+            var autoLogonCountValue = _registryManager.GetValue(RegistryHive.LocalMachine, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogon);
+            if (!string.IsNullOrEmpty(autoLogonCountValue))
+            {
+                warningReasons.Add(StringUtil.Loc("UITestingWarning_AutoLogonCount"));
+            }           
+            
+            return warningReasons;
+        }
+
+        private void UpdateMachineSpecificRegistrySettings(string domainName, string userName)
+        {
+            var hive = RegistryHive.LocalMachine;
+
+            // SetValue(RegistryHive targetHive, string subKeyName, string name, string value)
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonUserName, userName);
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonDomainName, domainName);
+
+            //this call is to take the backup of the password key if already exists as we delete the key in the next step
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonPassword, "");
+            _registryManager.DeleteValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonPassword);
+            
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonCount, "");
+            _registryManager.DeleteValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonCount);
+
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogon, "1");
+
+            SetValue(hive, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReason, "0");
+            SetValue(hive, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReasonUI, "0");
+
+            SetValue(hive, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeCaption, "");
+            SetValue(hive, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeText, "");
+        }
+
+        private void UpdateUserSpecificRegistrySettings(string domainName, string userName)
+        {
+            if (!_windowsServiceHelper.HasActiveSession(domainName, userName))
+            {
+                string securityId = _windowsServiceHelper.GetSecurityId(domainName, userName);
+                UpdateRegistrySettingsForDifferentUser(securityId);                
+            }
+            else
+            {
+                UpdateRegistrySettingsForCurrentUser();
```

Comment:
HasActiveSession was giving the logged in user info, it could be different from the user who is running the current process. Removed this code now because of the following:
1. Now we are asking for restart always.
2. Taking SID approach in all the cases. 

---
In reply to: [118595186](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118595186) [](ancestors = 118595186)